utility/get_ip_addrs.py

```diff
@@ -0,0 +1,57 @@
+""" Look at Ceph storage pods, return ip-addresses """
+import subprocess
+import shlex
+
+
+def get_ip_addrs_for_ceph_type(pod_type, namespace='openshift-storage'):
+    """
+    Get the ip address for all pods of a given type.
+    Args:
+        pod_type (string): pod type (mon, mgr, osd, all for everbody)
+        namespace (string): namespace in -n field of oc commands
+    Returns:
+        List of ip_addresses for all pods of the type specified
+    """
+    ret_list = []
+    look_for = f"rook-ceph-{pod_type}"
```

Comment:
I guess the prefix you used here (rook-ceph)is actually namespace, so may not be hard coded.

Reply:
```bash
rook-ceph-agent-8bgqb                         1/1     Running     0          6m28s
rook-ceph-agent-9zf27                         1/1     Running     0          6m28s
rook-ceph-agent-gmhwh                         1/1     Running     0          6m28s
rook-ceph-mgr-a-56f8b5d58b-77b7l              1/1     Running     0          3m5s
rook-ceph-mon-a-855f8b5bdd-mkss5              1/1     Running     0          4m33s
rook-ceph-mon-b-6f6cb4cd85-9h42g              1/1     Running     0          4m
rook-ceph-mon-c-5b9d4b9f8d-d79nf              1/1     Running     0          3m23s
rook-ceph-operator-7455674c54-6zd2p           1/1     Running     0          7m17s
rook-ceph-osd-0-84c7977546-cv77d              1/1     Running     0          2m15s
rook-ceph-osd-1-5bd66b959-x9w8v               1/1     Running     0          2m14s
rook-ceph-osd-2-59748b69f9-c97hm              1/1     Running     0          2m14s
rook-ceph-osd-prepare-ip-10-0-133-252-nmqn4   0/2     Completed   0          2m30s
rook-ceph-osd-prepare-ip-10-0-144-33-zzz6k    0/2     Completed   0          2m30s
rook-ceph-osd-prepare-ip-10-0-166-104-vrwkr   0/2     Completed   0          2m30s
rook-ceph-tools-75867b94fc-wmt6n              1/1     Running     0          5m8s
rook-discover-kq2zm                           1/1     Running     0          6m28s
rook-discover-lf42d                           1/1     Running     0          6m28s
rook-discover-wc95x                           1/1     Running     0          6m28s
```
We name the pods like this from templates I think.

But agree would be good to have this as parameter in function with default value "rook-ceph" used from defaults

Reply:
In downstream, rook-ceph prefix doesn't match namespace.

Reply:
I'll implement this as another parameter.

Reply:
I think that we can avoid any searching for a prefix.  A user can search for the text of the ceph component.  I don't think the user will care about the prefix.  (In the first use of this function the user is me).  Searching for 'ceph-mon' will return rook-ceph-mon-a, rook-ceph-mon-b, and rook-ceph-mon-c in the above output.